pkg/ovs/configjob.go

```diff
@@ -0,0 +1,120 @@
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ConfigJob - prepare job to configure ovn-controller
+func ConfigJob(
+	ctx context.Context,
+	h *helper.Helper,
+	k8sClient client.Client,
+	instance *v1beta1.OVS,
+	labels map[string]string,
+) ([]*batchv1.Job, error) {
+
+	var jobs []*batchv1.Job
+	runAsUser := int64(0)
+	privileged := true
+	jobTTLAfterFinished := int32(10)
+
+	dbmap, err := ovnclient.GetDBEndpoints(ctx, h, instance.Namespace, map[string]string{})
+	if err != nil {
+		return nil, err
+	}
+
+	ovsNodes, err := getOvsPodsNodes(
+		ctx,
+		k8sClient,
+		instance,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	envVars := map[string]env.Setter{}
+	envVars["KOLLA_CONFIG_FILE"] = env.SetValue(KollaConfigAPI)
+	envVars["KOLLA_CONFIG_STRATEGY"] = env.SetValue("COPY_ALWAYS")
+	envVars["OvnBridge"] = env.SetValue(instance.Spec.ExternalIDS.OvnBridge)
+	envVars["OvnRemote"] = env.SetValue(dbmap["SB"])
+	envVars["OvnEncapType"] = env.SetValue(instance.Spec.ExternalIDS.OvnEncapType)
+	envVars["PodNamespace"] = env.SetValue(instance.Namespace)
+	envVars["PodNetworksStatus"] = EnvDownwardAPI("metadata.annotations['k8s.v1.cni.cncf.io/networks-status']")
+	envVars["OvnEncapNetwork"] = env.SetValue(instance.Spec.NetworkAttachment)
+	envVars["OvnEncapIP"] = EnvDownwardAPI("status.podIP")
+	envVars["EnableChassisAsGateway"] = env.SetValue(fmt.Sprintf("%t", instance.Spec.ExternalIDS.EnableChassisAsGateway))
+	envVars["PhysicalNetworks"] = env.SetValue(getPhysicalNetworks(instance))
+	envVars["OvnHostName"] = EnvDownwardAPI("spec.nodeName")
+
+	for _, ovsNode := range ovsNodes {
+		//configHash := instance.Status.Hash[ovsv1beta1.OvnConfigHash+"-"+ovsNode]
+		//envVars["CONFIG_HASH"] = env.SetValue(configHash)
+		jobs = append(
+			jobs,
+			&batchv1.Job{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      instance.Name + "-configuration-" + ovsNode,
+					Namespace: instance.Namespace,
+					Labels:    labels,
+				},
+				Spec: batchv1.JobSpec{
+					TTLSecondsAfterFinished: &jobTTLAfterFinished,
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							RestartPolicy:      "OnFailure",
+							ServiceAccountName: ServiceAccountName,
+							Containers: []corev1.Container{
+								{
+									Name:  instance.Name + "-ovn-configuration-sync",
+									Image: instance.Spec.OvnContainerImage,
+									Command: []string{
+										"/usr/local/bin/container-scripts/init.sh",
+									},
+									Args: []string{},
+									SecurityContext: &corev1.SecurityContext{
+										Capabilities: &corev1.Capabilities{
+											Add:  []corev1.Capability{"NET_ADMIN", "SYS_ADMIN"},
+											Drop: []corev1.Capability{},
+										},
+										RunAsUser:  &runAsUser,
+										Privileged: &privileged,
+									},
+									Env:          env.MergeEnvs([]corev1.EnvVar{}, envVars),
+									VolumeMounts: GetOvnVolumeMounts(),
```

Comment:
(no action required) not all volumes here are really needed. e.g. we don't need a mount with /var/log/ovn. Probably others are not needed too. We can tackle it later when we merge operators.